src/lib/Core.jsx

```diff
@@ -150,30 +173,61 @@ const Core = function ({
 
     if (filteredValue !== 'all') {
       if (filteredValue === 'correct') {
-        filteredQuestions = questions.filter((question, index) => correct.indexOf(index) !== -1);
-        filteredUserInput = userInput.filter((input, index) => correct.indexOf(index) !== -1);
+        filteredQuestions = questions.filter(
+          (question, index) => correct.indexOf(index) !== -1,
+        );
+        filteredUserInput = userInput.filter(
+          (input, index) => correct.indexOf(index) !== -1,
+        );
       } else {
-        filteredQuestions = questions.filter((question, index) => incorrect.indexOf(index) !== -1);
-        filteredUserInput = userInput.filter((input, index) => incorrect.indexOf(index) !== -1);
+        filteredQuestions = questions.filter(
+          (question, index) => incorrect.indexOf(index) !== -1,
+        );
+        filteredUserInput = userInput.filter(
+          (input, index) => incorrect.indexOf(index) !== -1,
+        );
       }
     }
 
     return (filteredQuestions || questions).map((question, index) => {
-      const userInputIndex = filteredUserInput ? filteredUserInput[index] : userInput[index];
+      const userInputIndex = filteredUserInput
+        ? filteredUserInput[index]
+        : userInput[index];
 
       // Default single to avoid code breaking due to automatic version upgrade
       const answerSelectionType = question.answerSelectionType || 'single';
 
       return (
-        <div className="result-answer-wrapper" key={index + 1}>
-          <h3 dangerouslySetInnerHTML={rawMarkup(`Q${question.questionIndex}: ${question.question}`)} />
-          {question.questionPic && <img src={question.questionPic} alt="image" />}
-          {renderTags(answerSelectionType, question.correctAnswer.length, question.segment)}
-          <div className="result-answer">
-            {renderAnswerInResult(question, userInputIndex)}
+        <>
+          <div className="result-answer-wrapper" key={index + 1}>
+            <h3
+              dangerouslySetInnerHTML={rawMarkup(
+                `Q${question.questionIndex}: ${question.question}`,
+              )}
+            />
+            {question.questionPic && (
+              <img src={question.questionPic} alt="image" />
+            )}
+            {renderTags(
+              answerSelectionType,
+              question.correctAnswer.length,
+              question.segment,
+            )}
+            <div className="result-answer">
+              {renderAnswerInResult(question, userInputIndex)}
+            </div>
+            <Explanation question={question} isResultPage />
           </div>
-          <Explanation question={question} isResultPage />
-        </div>
+          <div>
+            {index === 5 ? (
```

Comment:
why it shows when the index is 5?

Reply:
It was when the last question was submitted then at the last page We have to add Restart button because there was five question starting from index 0
![Screenshot 2023-10-03 214846](https://github.com/wingkwong/react-quiz-component/assets/119872509/3ae4d2c2-a787-4077-8f4f-9ddf123a190e)

Reply:
Why not just add it at the end? You can't assume there is only 5 questions each time as the number varies.